### PR TITLE
Warning suppression for C-API.

### PR DESCRIPTION
**Pull request by xipingyan**
Signed-off-by: Yan, Xiping <xiping.yan@intel.com>

### Details:
 - *Adding warnings suppression for C API*
 

### Tickets:
 - *89774*

### CODE REVIEW

**Review by ilyachur — COMMENTED** [review]
I think we also need to have some changes in CMake in order to validate these changes in our CI.

File: src/bindings/c/CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 project(OpenVINO_C_API)
 
+if(CMAKE_COMPILER_IS_GNUCXX)
```
Comment (ilyachur) [inline src/bindings/c/CMakeLists.txt:7]: I think you can just change the order of components in common CMake script.
As it was done in PR https://github.com/openvinotoolkit/openvino/pull/12666

Comment (ilya-lavrenov) [reply src/bindings/c/CMakeLists.txt:7]: in this case they have to fix python as well. I suppose we can merge as is and remove these options later

Comment (xipingyan) [reply src/bindings/c/CMakeLists.txt:7]: Yes, let's update it when we merge https://github.com/openvinotoolkit/openvino/pull/12666.

Comment (ilya-lavrenov) [reply src/bindings/c/CMakeLists.txt:7]: @xipingyan I've merged my PR, please proceed with this one

Comment (xipingyan) [reply src/bindings/c/CMakeLists.txt]: Done, please help review again.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -19,6 +19,8 @@ endif()
 if(CMAKE_COMPILER_IS_GNUCXX)
     ie_add_compiler_flags(-Wall)
     ie_add_compiler_flags(-Wmissing-declarations)
+    ie_add_compiler_flags(-Wno-error=sign-compare)
+    ie_add_compiler_flags(-Wno-error=missing-declarations)
```
Comment (ilyachur) [inline src/CMakeLists.txt:23]: Why did toy relax these warnings?
I thought that you should fix these issues in this PR

Comment (xipingyan) [reply src/CMakeLists.txt]: Sorry, forget to remove no-error.
Just updated, I move sign-compare to below, because plugins have many compare errors.

Comment (xipingyan) [reply src/CMakeLists.txt]: Hi @ilyachur , do we need to create new JIRA to track plugins compara error? or I update these error in this PR? What do you think?

Comment (ilyachur) [reply src/CMakeLists.txt]: I am not expect that on this PR you will found new issues inside plugins

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -26,6 +26,10 @@ add_subdirectory(inference)
 include(cmake/openvino.cmake)
 # preprocessing has dependency on `openvino` for static build
 add_subdirectory(common/preprocessing)
+
+if(CMAKE_COMPILER_IS_GNUCXX)
```
Comment (ilyachur) [inline src/CMakeLists.txt:30]: Why did you add this additional flag?

Comment (xipingyan) [reply src/CMakeLists.txt:30]: Actually I am not sure whether we need to add "-Wsign-compare", if not, I will remove it.

**Review by ilyachur — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]
